### PR TITLE
Stop using globs for `chokidar` watchers

### LINKED ISSUE #12241: Update file watching for `chokidar` version 4
In the last release, `chokidar` (the file system watcher) was upgraded from version 3 to version 4. The update includes some breaking changes. The most important one is casually mentioned in that it "removes support for globs".

This seems to break the CesiumJS processes in some places - for example, the [watching mechanism in `server.js`](https://github.com/CesiumGS/cesium/blob/e3e98371830a85d653c2f9e1f34f382c84984bce/server.js#L219), which passes an array of `sourceFiles` (which contain glob expressions) to chokidar.

There are some hints for upgrading in https://github.com/paulmillr/chokidar#upgrading , but one has to be careful to replicate the original behavior (for example, to ensure that it does not only watch _changed_ files, but also picks up _new_ files and such...)

### PR DESCRIPTION
**Pull request by jjspace**
# Description

`chokidar` v4 removed support for file globs in it's watchers. This restructures how we choose what files to watch to use the new API. The functionality of the local server should not have changed so please confirm this.

The new system recursively watches files if given a directory path so the old `file/path/**/*` pattern is exactly the same as `file/path` in the new system.

The new system relies much more on the `ignored` option to filter _out_ files that you don't want it to watch. This creates a bit of separation between the list of files we want and files we don't want which means we can't just set one array of globs at the top of the file anymore.

## Issue number and link

Fixes #12241 

## Testing plan

- Run the local server `npm start`
    - Optionally add a `console.log` statement in each of the 3 `watcher.on()` handlers to confirm when they fire
- Check that each of the following events triggers a re-build as expected in the directories we specify, like Source and Specs
    - Create a file (`touch test.js`)
    - Delete a file (`rm test.js`)
    - Modify a file (save in vscode)
- Check that the watchers are _not_ triggered for those events in directories we want ignored
- Verify everything on localhost actually still works as expected

# Author checklist

- [ ] I have submitted a Contributor License Agreement
- [ ] I have added my name to `CONTRIBUTORS.md`
- [ ] I have updated `CHANGES.md` with a short summary of my change
- [ ] I have added or updated unit tests to ensure consistent code coverage
- [ ] I have updated the inline documentation, and included code examples where relevant
- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by javagl** [comment]
> Check that each of the following events triggers a re-build as expected in the directories we specify, like Source and Specs
> Create a file (touch test.js)

Is this _supposed_ to work?

When adding a `CartesianX.js`, it bails out with
```
X [ERROR] No matching export in "packages/engine/index.js" for import "CartesianX"

    Source/Cesium.js:221:9:
      221 │ export { CartesianX } from '@cesium/engine';
          │          ~~~~~~~~~~
          ╵          Cartesian4

  Did you mean to import "Cartesian4" instead?

    packages/engine/index.js:171:9:
      171 │ export { default as Cartesian4 } from './Source/Core/Cartesian4.js';
          ╵          ~~~~~~~

1 error
Error: Build failed with 1 error:
Source/Cesium.js:221:9: ERROR: No matching export in "packages/engine/index.js" for import "CartesianX"
```
(similarly, when restarting and then _deleting_ it, this also causes an error)

# but...

the same happens for me with `"chokidar": "^3.5.3",`, so ... I think that this may _never_ have worked? 

I'm rarely adding/removing files. And I think that it happens rarely enough _in general_ to not consider this as an issue.

---

The main poins:

> Modify a file (save in vscode)

I tried it out with _modifications_ in a source file, and this worked and was picked up properly. 

(I can do additional tests with shaders, but there, https://github.com/CesiumGS/cesium/issues/12164 might interfere...)

> Check that the watchers are not triggered for those events in directories we want ignored

I added/deleted a file in one of the `ignored` source directories, and this was ignored.

> Verify everything on localhost actually still works as expected

"Everything" is a bit too much for me to verify right now 😄 but I didn't see any issues.

**Review by javagl — APPROVED** [review]
I did try it out, as summarized in the comment above. But I'm not sooo deeply familiar with the build processes and tooling. So I can approve the changes **from my limited perspective**. 

But I hope that nobody will blame me when something goes wrong in some deeper build step during the next release...

File: server.js [hunk 1]
```diff
@@ -46,27 +46,6 @@ import {
   buildCesium,
 } from "./scripts/build.js";
 
-const sourceFiles = [
-  "packages/engine/Source/**/*.js",
-  "!packages/engine/Source/*.js",
-  "packages/widgets/Source/**/*.js",
-  "!packages/widgets/Source/*.js",
```
[the remark this answers is not in the document]
Comment (javagl) [reply server.js:53]: There don't seem to be matching files in the `engine` part. But I wonder 

- whether it _previously_ ignored changes in files like `packages/widgets/Source/ClockViewModel.js` that will now no longer be ignored. 
- whether these files _should_ be ignored. (I don't see a reason to ignore them, but ... what do I know...)

**Comment by ggetz** [comment]
> > Check that each of the following events triggers a re-build as expected in the directories we specify, like Source and Specs
> Create a file (touch test.js)

> Is this supposed to work?

> When adding a CartesianX.js, it bails out with

This doesn't (and didn't even before the chokidar version update) because we are only calling `createCesiumJs`. It updates the top-level `Source/Cesium.js` file, but not the individual `package/*/index.js` files. We need to also call `createIndexJs` for the relevant workspace. 

It a small change that I'll make now.

**Comment by ggetz** [comment]
Thanks @jjspace and @jjspace!

**Pull request merged**